File: examples/swift/language_gauntlet/Sources/LanguageGauntlet/Domain/Envelope.swift
public enum Kind {
    case run
    case eval
}

public struct Envelope {
    public var kind: Kind
    public var cmd: String
    public var user: String
    public var length: Int
    public var extras: [String]

    public init(kind: Kind, cmd: String, user: String, length: Int, extras: [String]) {
        self.kind = kind
        self.cmd = cmd
        self.user = user
        self.length = length
        self.extras = extras
    }
}

File: examples/java/language_gauntlet/src/main/java/gauntlet/domain/Envelope.java
package gauntlet.domain;

import java.util.List;

public record Envelope(Kind kind, String cmd, String user, int length, List<String> extras) {
    public enum Kind { RUN, EVAL }

    public static Envelope fromRequest(String cmd, String user) {
        return new Envelope(Kind.RUN, cmd, user, cmd.length(), List.of(cmd));
    }
}
